### PR TITLE
[Fix](point query) Fix point query unstable

### PR DESCRIPTION
**Pull request by Yukang-Lian**
## Proposed changes

Issue Number: close #xxx

Point queries can fluctuate in both MoR and MoW, mainly because multiple segments are merged to generate a new segment. After the merge, new queries will read the new segment, but the new segment is not in the cache. It needs to do extra disk I/O and some calculations. When the number of new segments read is large, it will cause a large delay. The solution is simple: add the new segment to the segment cache after each compaction.

To reproduce the process, first import a portion of the data for a wide table with a large amount of base data, and then perform a query after the compaction is complete.

Before this PR, it is easy to reproduce.

Benchmark
        Average number of seconds to run all queries: 0.010 seconds
        Minimum number of seconds to run all queries: 0.007 seconds
        Maximum number of seconds to run all queries: 0.112 seconds

After this PR, we do this operation nearly 100 times, the unstable phenomenon never happens.

Benchmark
        Average number of seconds to run all queries: 0.008 seconds
        Minimum number of seconds to run all queries: 0.007 seconds
        Maximum number of seconds to run all queries: 0.016 seconds

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by Yukang-Lian** [comment]
run buildall

File: be/src/olap/cumulative_compaction.cpp [hunk 1]
```diff
@@ -95,6 +97,18 @@ Status CumulativeCompaction::execute_compact_impl() {
     VLOG_CRITICAL << "after cumulative compaction, current cumulative point is "
                   << _tablet->cumulative_layer_point() << ", tablet=" << _tablet->full_name();
 
+    if (_tablet->enable_unique_key_merge_on_write()) {
```
Comment (liaoxin01) [inline be/src/olap/cumulative_compaction.cpp:100]: duplicated code. Implement this function in the base class？

**Comment by Yukang-Lian** [comment]
run buildall

**Comment by Yukang-Lian** [comment]
run buildall

**Review by liaoxin01 — APPROVED** [review]
LGTM

**Comment by liaoxin01** [comment]
Add some description for this PR.

**Comment by Yukang-Lian** [comment]
run buildall

**Comment by wm1581066** [comment]
Is there any test data?

**Comment by Yukang-Lian** [comment]
> Is there any test data?

You can see it in the depiction.

**Review by zhannngchen — APPROVED** [review]
LGTM

**Pull request merged**